### DISCUSSION TITLE
Make HttpContext expose Node HTTP request object as well

### DISCUSSION THREAD

**Discussion by seivan** [adonisjs/core#4495] [Ideas]
More specifically [http.IncomingMessage](https://nodejs.org/dist/latest-v19.x/docs/api/http.html#class-httpincomingmessage) just like how  
[ServerResponse](https://nodejs.org/dist/latest-v19.x/docs/api/http.html#class-httpserverresponse) is [exposed under response](https://docs.adonisjs.com/guides/response#accessing-nodejs-res-object).

Hiding those makes integration harder with other systems.

For instance, exposing them would make things like adding graphql integration in a route a breeze.

```ts
import http from 'http';
import { createHandler } from 'graphql-http/lib/use/http';


const handler = createHandler({ schema });

// Create a route on `/graphql`
router.post('/graphql', ({ req, res }: HttpContext) => {
//missing nodeRequest
    handler(req.nodeRequest, res.response);
});
```

**Comment by Julien-R44** [comment]
They are already exposed by HttpContext : 

```ts
router.post('/graphql', ({ request, response }: HttpContext) => {
    // IncomingMessage and ServerResponse
    handler(req.request.request, res.response.response);
});
```

**Reply by seivan** [reply]
Oh cool, documentation didn’t mention that for request. Thank yoy.
